cmd/ulxly/bridge_service/agglayer/service.go

```diff
@@ -0,0 +1,262 @@
+package agglayer
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/0xPolygon/polygon-cli/cmd/ulxly/bridge_service"
+	"github.com/0xPolygon/polygon-cli/cmd/ulxly/bridge_service/httpjson"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/rs/zerolog/log"
+)
+
+const urlPath = "bridge/v1"
+
+type BridgeService struct {
+	bridge_service.BridgeServiceBase
+	httpClient *http.Client
+}
+
+// NewBridgeService creates an instance of the BridgeService.
+func NewBridgeService(url string, insecure bool) (*BridgeService, error) {
+	return &BridgeService{
+		BridgeServiceBase: bridge_service.NewBridgeServiceBase(url),
+		httpClient:        httpjson.NewHTTPClient(insecure),
+	}, nil
+}
+
+func (s *BridgeService) GetDeposit(depositNetwork, depositCount uint32) (*bridge_service.Deposit, error) {
+	bridgeEndpoint := fmt.Sprintf("%s/%s/bridges?network_id=%d&deposit_count=%d", s.BridgeServiceBase.Url(), urlPath, depositNetwork, depositCount)
+	bridgeResp, bridgeRespError, statusCode, err := httpjson.HTTPGetWithError[getBridgesResponse, errorResponse](s.httpClient, bridgeEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	if statusCode != http.StatusOK {
+		errMsg := "unable to retrieve bridge deposit"
+		log.Warn().Int("code", statusCode).Str("message", bridgeRespError.Error).Msgf("%s", errMsg)
```

Comment:
There is a risk of a panic here... if `statusCode` is not okay, I don't think there is a guarantee that `bridgeRespError.Error` is populated?